lib/puppet/parser/functions/ipaddresses.rb

```diff
@@ -1,41 +0,0 @@
-module Puppet::Parser::Functions
-  newfunction(:ipaddresses, type: :rvalue, doc: <<-EOS
-  Returns all ip addresses of network interfaces (except lo) found by facter.
-  Special network interfaces (e.g. docker0) can be excluded by an exclude list as
-  first argument for this function.
-EOS
-             ) do |args|
-    networking = lookupvar('networking')
-
-    # always exclude loopback interface
-    interfaces_exclude = ['lo']
-    if args.size == 1
-      unless args[0].is_a?(Array)
-        raise(Puppet::ParseError, 'ipaddresses(): Requires first argument to be an Array')
-      end
-      interfaces_exclude << args[0]
-    end
-
-    return false unless networking.include?('interfaces')
```

Comment:
I left this out as I couldn't figure out how the `networking` fact could exist, but it not have an `interfaces` key.